OKTA-529773 XaaS integration guide (#3630)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/reference/api/xaas/index.md

| `identitySourceId`  | String | The identity source ID obtained from creating a Custom Identity Source integration in Okta |
| `sessionId`  | String | The ID for the Identity Source Session to load deactivation user data |

#### Request body

| Property    | Type           | Description   |
| ----------- | -------------- | ------------- |
| `entityType` | String (enum: `USERS`) | The type of data to bulk-delete in a session. Currently, only `USERS` is supported.  |
| `profiles` | Array | An array of [Identity Source User Profile For Delete object](#identity-source-user-profile-for-delete-object) |

#### Response body

None

##### Possible errors

See [Possible common errors](#possible-common-errors) for all Identity Sources API endpoints. The following are possible errors specific to this endpoint:

| Error code  | HTTP code    | Description    |
| ----------- | -----------  | -------------- |
| `E0000003` | 400 Bad Request | There is no payload in the bulk-delete request. |
| `E0000001` | 400 Bad Request | The `profiles` array is missing or empty in the bulk-delete request. |
| `E0000003` | 400 Bad Request | The `entityType` property value is invalid. |

#### Use example

> This request loads a set of external IDs for user deactivation in an Identity Source Session with an `id` value of `${sessionId}`

Reviewer: ```suggestion
This request loads a set of external IDs for user deactivation in an Identity Source Session with an `id` value of `${sessionId}`:
```

##### Possible errors

See [Possible common errors](#possible-common-errors) for all Identity Sources API endpoints. The following are possible errors specific to this endpoint:

| Error code  | HTTP code    | Description    |
| ----------- | -----------  | -------------- |
| `E0000001` | 400 Bad Request | The Identity Source Session was already triggered and doesn't have the `CREATED` status. |

#### Use example

This request triggers the data import process for an Identity Source Session with an `id` value of `${sessionId}`:

##### Request

```bash
curl -X POST 'https://${yourOktaDomain}/api/v1/identity-sources/${identitySourceId}/sessions/${sessionId}/start-import' \
-H 'accept: application/json' \
-H 'authorization: SSWS ${apiToken}' \
-H 'cache-control: no-cache' \
-H 'content-type: application/json'
```

##### Response

```json
{
  "id": "{sessionId}",
  "identitySourceId": "{identitySourceId}",
  "status": "TRIGGERED",
  "importType": "INCREMENTAL"
}
```

### Cancel an Identity Source Session

<ApiOperation method="delete" url="/api/v1/identity-sources/${identitySourceId}/sessions/${sessionId}" />

> Deletes all the loaded bulk data in an Identity Source Session and cancels the session for further activity. Only Identity Source Sessions with the `CREATED` status can be canceled.

Reviewer: ```suggestion
Deletes all the loaded bulk data in an Identity Source Session and cancels the session for further activity. You can only cancel Identity Source Sessions with the `CREATED` status.
```

### Identity Source Session object

#### Identity Source Session properties

| Property    | Type           | Description   |
| ----------- | -------------- | ------------- |
| `id` | String | Unique identifier for the Identity Source Session |
| `identitySourceId` | String | Unique identifier obtained from creating a Custom Identity Source integration in Okta |
| `status` | String (enum: `CREATED`, `TRIGGERED`, `COMPLETED`, `CLOSED`, `EXPIRED`) | The current status of the Identity Source Session:<br><ul><li>`CREATED`: This is a new Identity Source Session that hasn't been processed. You can upload bulk data in this stage.</li><li>`TRIGGERED`: Okta is processing the import data in this session. You can't load bulk data in this stage.</li><li>`COMPLETED`: The bulk data was processed and imported into Okta.</li><li>`CLOSED`: The Identity Source Session was canceled and isn't available for further activity.<li>`EXPIRED`: This Identity Source Session had the `CREATED` status and timed-out after 24 hours of inactivity.</li></ul>|
| `importType` | String (enum: `INCREMENTAL`) | The import type for the Identity Source Session. Currently, only `INCREMENTAL` is supported. |

> #### Identity Source Session example

Reviewer: ```suggestion

#### Identity Source Session example
```
